bokeh/models/sources.py

```diff
@@ -38,6 +39,10 @@ class DataSource(PlotObject):
                 hit/selected
     """)
 
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the selection is changed.
+    """)
+
```

Comment:
I'm not opposed to adding more meanings to this callback later, but I don't think we can realistically do more with this PR and get it into 0.9.